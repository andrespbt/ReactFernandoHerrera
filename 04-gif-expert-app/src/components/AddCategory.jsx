import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
	const [value, setValue] = useState('');
	const onInputChange = ({ target: { value } }) => {
		setValue(value);
	};
	const handleSubmit = e => {
		e.preventDefault();

		if (value.trim() <= 1) return;

		onNewCategory(value);
		setValue('');
	};

	return (
		<form
			onSubmit={handleSubmit}
			aria-label="form">
			<input
				type="text"
				placeholder="Search Gifs"
				onChange={onInputChange}
				value={value}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired,
};
