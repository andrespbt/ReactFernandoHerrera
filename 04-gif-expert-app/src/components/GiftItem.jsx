import PropTypes from 'prop-types';

export const GiftItem = ({ title, url }) => {
	return (
		<div className="card">
			<img
				src={url}
				alt={title}
				height="275"
				width="275"
			/>
			<p>{title}</p>
		</div>
	);
};

GiftItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};

/* Tarea 

	1.AÃ±adir proptypes

		.Title obligatorio
		.url obligatorio

	2. Evaluar snapshot


*/
