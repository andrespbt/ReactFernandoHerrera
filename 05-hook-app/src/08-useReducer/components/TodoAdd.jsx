import { useState } from 'react';

export const TodoAdd = ({ handleNewTodo }) => {
	const [value, setValue] = useState('');

	const onInputChange = ({ target: { value } }) => {
		setValue(value);
	};

	const onNewTodo = e => {
		e.preventDefault();

		if (!value.trim().length > 1) {
			return;
		}

		handleNewTodo({
			id: new Date().getTime(),
			description: value,
			done: false,
		});

		setValue('');
	};

	return (
		<form
			onSubmit={onNewTodo}
			aria-label="form">
			<input
				type="text"
				placeholder="¿Qué hay que hacer?"
				className="from-control"
				onChange={onInputChange}
				value={value}
			/>

			<button
				type="submit"
				className="btn btn-outline-primary mt-1">
				Agregar
			</button>
		</form>
	);
};
