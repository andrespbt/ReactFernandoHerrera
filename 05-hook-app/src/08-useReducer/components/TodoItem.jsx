export const TodoItem = ({ todo: { id, done, description }, onDeleteTodo, onToggleTodo }) => {
	return (
		<li
			key={id}
			className="list-group-item d-flex justify-content-between">
			<span
				className={`align-self-center ${done ? 'text-decoration-line-through' : ''}`}
				onClick={() => onToggleTodo(id)}
				aria-label="span">
				{description}
			</span>
			<button
				className="btn btn-danger"
				onClick={() => onDeleteTodo(id)}>
				Borrar
			</button>
		</li>
	);
};
