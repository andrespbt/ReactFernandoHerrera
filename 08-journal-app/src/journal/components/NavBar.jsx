import { useDispatch, useSelector } from 'react-redux';
import { MenuOutlined, LogoutOutlined } from '@mui/icons-material';
import { AppBar, Grid, IconButton, Toolbar, Typography } from '@mui/material';
import { startLogOut } from '../../store/auth/thunks';
export const Navbar = ({ drawerWidth = 240 }) => {
  const dispatch = useDispatch();

  const onLogout = () => {
    dispatch(startLogOut());
  };

  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}>
      <Toolbar>
        <IconButton
          color="inherit"
          edge="start"
          sx={{ mr: 2, display: { sm: 'none' } }}>
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          direction="row"
          justifyContent="space-between"
          align-items="center">
          <Typography
            variant="h6"
            noWrap
            component="div">
            Journal App
          </Typography>

          <IconButton
            color="error"
            onClick={onLogout}>
            <LogoutOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
