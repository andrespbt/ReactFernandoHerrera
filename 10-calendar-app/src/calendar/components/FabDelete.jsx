import { useCalendarStore } from '../../hooks';

export const FabDelete = () => {
  const { startDeletingEvent, hasEventSelected, activeEvent } = useCalendarStore();

  const handleDelete = () => {
    startDeletingEvent(activeEvent);
  };

  return (
    <button
    aria-label='btnDelete'
      className="btn btn-danger fab-danger"
      onClick={handleDelete}
      style={{
        display: hasEventSelected ? '' : 'none',
      }}
    >
      <i className="fas fa-trash-alt"></i>
    </button>
  );
};
