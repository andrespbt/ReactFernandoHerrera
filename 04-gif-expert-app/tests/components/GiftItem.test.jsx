import { render, screen } from '@testing-library/react';
import { GiftItem } from '../../src/components/GiftItem';

describe('Test on GiftItem', () => {
	const title = 'Goku';
	const url = 'https://goku.com/';

	test('should match the snapshot', () => {
		const { container } = render(
			<GiftItem
				title={title}
				url={url}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	test('should show the image with url and alt correctly', () => {
		render(
			<GiftItem
				title={title}
				url={url}
			/>
		);

		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('should show title in component', () => {
		render(
			<GiftItem
				title={title}
				url={url}
			/>
		);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
