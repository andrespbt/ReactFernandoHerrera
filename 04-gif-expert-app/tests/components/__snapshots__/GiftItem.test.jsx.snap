// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      height="275"
      src="https://goku.com/"
      width="275"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
